src/hooks/useTransactionHistory.tsx

```diff
@@ -0,0 +1,245 @@
+      let principalSymbol = '';
+      let collateralName = '';
+      let principalName = '';
+      let symbol = '';
+      let name = '';
+
+      if (hasCollateralReserve(txn)) {
+        collateralSymbol = txn.collateralReserve.symbol.toLowerCase();
+        collateralName = txn.collateralReserve.name.toLowerCase();
+      }
+
+      if (hasPrincipalReserve(txn)) {
+        principalSymbol = txn.principalReserve.symbol.toLowerCase();
+        principalName = txn.principalReserve.name.toLowerCase();
+      }
+
+      if (hasReserve(txn)) {
+        symbol = txn.reserve.symbol.toLowerCase();
+        name = txn.reserve.name.toLowerCase();
+      }
+
+      // handle special case where user searches for ethereum but asset names are abbreviated as ether
+      const altName = name.includes('ether') && !name.includes('tether') ? 'ethereum' : '';
+
+      return (
+        symbol.includes(lowerSearchQuery) ||
+        collateralSymbol.includes(lowerSearchQuery) ||
+        principalSymbol.includes(lowerSearchQuery) ||
+        name.includes(lowerSearchQuery) ||
+        altName.includes(lowerSearchQuery) ||
+        collateralName.includes(lowerSearchQuery) ||
+        principalName.includes(lowerSearchQuery)
+      );
+    });
+  } else {
+    filteredTxns = txns;
+  }
+
+  // apply txn type filter
+  if (filterQuery.length > 0) {
+    filteredTxns = filteredTxns.filter((txn: TransactionHistoryItemUnion) => {
+      if (filterQuery.includes(actionFilterMap(txn.action))) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+  }
+  return filteredTxns;
+};
+
+export const useTransactionHistory = ({ isFilterActive }: { isFilterActive: boolean }) => {
+  const [currentMarketData, account] = useRootStore((state) => [
+    state.currentMarketData,
+    state.account,
+  ]);
+
+  const [shouldKeepFetching, setShouldKeepFetching] = useState(false);
+
+  // Handle subgraphs with multiple markets (currently only ETH V2 and ETH V2 AMM)
+  let selectedPool: string | undefined = undefined;
+  if (
+    !currentMarketData.v3 &&
+    (currentMarketData.marketTitle === 'Ethereum' ||
+      currentMarketData.marketTitle === 'Ethereum AMM')
+  ) {
+    selectedPool = currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER.toLowerCase();
+  }
+
+  interface TransactionHistoryParams {
+    account: string;
+    subgraphUrl: string;
+    first: number;
+    skip: number;
+    v3: boolean;
+    pool?: string;
+  }
+  const fetchTransactionHistory = async ({
+    account,
+    subgraphUrl,
+    first,
+    skip,
+    v3,
+    pool,
+  }: TransactionHistoryParams) => {
+    let query = '';
+    if (v3) {
+      query = USER_TRANSACTIONS_V3;
+    } else if (pool) {
+      query = USER_TRANSACTIONS_V2_WITH_POOL;
+    } else {
+      query = USER_TRANSACTIONS_V2;
+    }
+
+    const requestBody = {
+      query,
+      variables: { userAddress: account, first, skip, pool },
+    };
+    try {
+      const response = await fetch(subgraphUrl, {
```

Comment:
Not a huge deal, but if a market doesn't have a subgraphUrl, it just shows the loading skeleton indefinitely. Only an issue for Scroll and Sepolia test nets right now. Maybe we could just show some message that history is not available if there's no subgraph url configured?

Reply:
Good point, and that actually reminded me that all of the V3 testnet subgraphs needed to be disabled because the subgraphs do not match UI markets which were re-deployed w/ the permissioned faucet update. I added a placeholder for these markets:

![TxHistoryUnavailable](https://github.com/aave/interface/assets/26660211/8170ae6f-5ba7-4ce5-989d-ede6057da64c)